### ISSUE TITLE
/vagrant default share has no override_options

### ISSUE THREAD

**Issue by yareckon** [geerlingguy/drupal-vm#1375]
### Issue Type

 - Bug Report / Support Request

### Your OS

 - **Linux** (Debian Stretch)

### Summary

DrupalVM keeps the default /vagrant share inside the VM, allowing it to pick up the user's configured default type (https://github.com/geerlingguy/drupal-vm/blob/master/Vagrantfile#L95).  There is no chance as far as I can see to add override_options like the custom mounts from the config.yml to the default share.  Why this is important is for instance if you need to switch to nfs version 4 or tcp rather than udp nfs.  Could you a way to either disable or configure the default /vagrant share?  I'm listing this as a bug, because vagrant will break if you have no way to change incompatible mount options.

Thanks for a great project!

**Comment by oxyc** [comment]
You can also set this using a `Vagrantfile.local` for now:

```rb
config.vm.synced_folders['/vagrant'][:nfs_version] = 4
config.vm.synced_folders['/vagrant'][:nfs_udp] = false
```

**Comment by yareckon** [comment]
Thank you for that suggestion -- that looks like my solution for now!

For anyone reaching this issue late, heres' the current (mid-2017) drupalvm doc page on Vagrantfile.local http://docs.drupalvm.com/en/2.5.1/other/local-vagrantfile/

I would still question whether a default mount not visible in config.yml should exist in drupalvm by default.  If it does for reasons of not rocking the vagrant boat, we should at least be able to manipulate its options in the same config files as everything else.

**Comment by geerlingguy** [comment]
> If it does for reasons of not rocking the vagrant boat

That's the main reason, yes.

> We should at least be able to manipulate its options in the same config files as everything else.

Yeah, probably so. One question, though—was it causing provisioning to fail entirely, or was there another reason you needed to change the mount options? (Just want to document an error if there was one so google can pick up this issue for other Linux users).

**Comment by yareckon** [comment]
It was about being able to work on the machine after vagrant up.  After typing in host passwords to mount nfs, there was a hang for about 5 minutes until nfs guest gave up trying to mount.

**Comment by oxyc** [comment]
Related https://github.com/mitchellh/vagrant/issues/2304 and https://www.vagrantup.com/docs/synced-folders/nfs.html#other-notes

**Comment by geerlingguy** [comment]
Fixed via just-merged PR #1392
